### PR TITLE
Allow to specify the port for a ssh connection

### PR DESCRIPTION
**Pull request by dcermak**
Currently ssh connections always use port 22. But for vagrant + virtualbox we
must be able to change this to a different value, as this backend uses ports
2200 and higher.
For that we extend `get_ssh_credentials` to also allow to return a `port` hash
key which is then forwarded to `ssh->connect()`.

### CODE REVIEW

File: backend/baseclass.pm [hunk 1]
```diff
@@ -1270,8 +1271,10 @@ sub new_ssh_connection {
 
     # Retry multiple times, in case of the guest is not running yet
     my $counter = $bmwqemu::vars{SSH_CONNECT_RETRY} // 5;
+    my $con_pretty = "$args{username}\@$args{hostname}";
+    $con_pretty = $con_pretty . ":$args{port}" unless $args{port} == 22;
```
Comment (Martchus) [inline backend/baseclass.pm:1275]:
```suggestion
    $con_pretty = "$con_pretty:$args{port}" unless $args{port} == 22;
```

**Review by Martchus — COMMENTED** [review]
Looks like tidy doesn't like it.

**Review by Martchus — APPROVED** [review]

**Comment by dcermak** [comment]
> Looks like tidy doesn't like it.

I'll try to fix that and will also add some tests.

**Comment by dcermak** [comment]
Turns out I actually had a bug in there with `keep_alive => 1`: you wouldn't get a new connection when connecting to the same host at different ports…

File: backend/baseclass.pm [hunk 2]
```diff
@@ -1269,9 +1270,11 @@ sub new_ssh_connection {
     my $ssh = Net::SSH2->new;
 
     # Retry multiple times, in case of the guest is not running yet
-    my $counter = $bmwqemu::vars{SSH_CONNECT_RETRY} // 5;
+    my $counter    = $bmwqemu::vars{SSH_CONNECT_RETRY} // 5;
+    my $con_pretty = "$args{username}\@$args{hostname}";
+    $con_pretty = "$con_pretty:$args{port}" unless $args{port} == 22;
```
Comment (okurz) [inline backend/baseclass.pm:1275]:
```suggestion
    $con_pretty .= ":$args{port}" unless $args{port} == 22;
```

**Review by okurz — CHANGES_REQUESTED** [review]

**Pull request merged**